src/new_ir.jl

```diff
@@ -145,6 +166,39 @@ function generate(t::NewAbstractTerm,params...)
     itr = [ name(p)=>Symbol("arg$i") for (i,p) in enumerate(params)  ]
     symbols = map(last,itr)
     name_to_symbol = Dict(itr)
+    
+    # find all leaf terms 
+    name_to_captured_data = Dict()
+    function capture!(a::ContributionTerm)
```

Comment:
Can we move these functions outside? Otherwise this will grow up a lot when adding new term types. BTW, this looks similar to the get_symbol! function that we already have. Take inspiration from it if needed.